### ISSUE TITLE
GARD problem

### ISSUE THREAD

**Issue by mdbohorqueza** [veg/hyphy#1219]
Hello:
When I try to run GARD, it is showing me this message that there was an error processing my job. I have run this same data set before and I was just verifying.
I had to run as a hurry on classic server but I do not know if it would be the same result.
https://www.datamonkey.org/gard/5f5a63c6dc5d1378a560e7a8

**Comment by stephenshank** [comment]
Dear @mdbohorqueza,

Thanks for reporting this issue. HyPhy's GARD analysis recently underwent a refactoring, and it seems to be experiencing issues instantiating codon models with rate variation. For example, when I run:

```
mpirun -np 4 HYPHYMPI gard --type codon --alignment Flu.fasta --rv Gamma
```

I see the error 

```
Master node received an error:User-defined function 'models.codon.MG_REV.ModelDescription' needs 2 parameters, but 1 were supplied 
```

but this disappears when I remove the rate variation argument. This is the same error I see when downloading your error log, and I also see that you selected a Gamma distribution with 5 bins.

I will continue to look into this and discuss with other lab members, and follow up when I have more to report.

Best,
Stephen

**Comment by mdbohorqueza** [comment]
Thank you stephenshank
I have run it with different rate variation. First I tried none and General discrete 3 as it is recomended, but I decided to try several rates because I noted that it was having too much influence on the results. Here are some examples:
https://www.datamonkey.org/fel/5f5a4b19dc5d1378a560e1a6#tree-tab
https://www.datamonkey.org/fel/5f5aa87cdc5d1378a560f09f
Finally I decided try 5 bins because of the large differences between runs and the Jmodeltest results that showed evolutionary rate differences among sites with 5 categories. 
Now I was verifying the results but because I was not sure what parameter I made the final analysis I decided to run it again. However, currently it is resulting the same problem no matter the parameters used.
Thanks for yur help.
